hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBlobClient.java

```diff
@@ -325,11 +330,31 @@ public AbfsRestOperation createPath(final String path,
       final boolean isAppendBlob,
       final String eTag,
       final ContextEncryptionAdapter contextEncryptionAdapter,
+      final boolean isRecursiveCreate,
       final TracingContext tracingContext, final boolean isNamespaceEnabled)
       throws AzureBlobFileSystemException {
-    return createPath(path, isFile, overwrite, permissions, isAppendBlob, eTag,
-        contextEncryptionAdapter, tracingContext, isNamespaceEnabled, false);
+    AbfsLease abfsLease = null;
```

Comment:
I might be missing something...
Can you help me recall why we need this lease business now and not earlier??

Or was it just missed earlier?

Reply:
Yet it was missed. In case of createNonRecursive on atomic path, we have to take lease on the parent directory.